### PR TITLE
if Windows, append .exe to binaryName in discover(), to support mingw64 and cygwin compilers

### PR DESCRIPTION
**Pull request by philwalk**
This change enables discovery of `clang.exe` and `clang++.exe`,  in these environments:
```
CYGWIN_NT-10.0 d5 3.2.0(0.340/5/3) 2021-03-29 08:42 x86_64 Cygwin
MSYS_NT-10.0-19042 d5 3.2.0-340.x86_64 2021-07-02 08:36 UTC x86_64 Msys
```
Also verified that discovery still works when first `clang` in the PATH is here:
```
C:/Program Files (x86)/Microsoft Visual Studio/2019/Preview/VC/Tools/Llvm/x64
```

### CODE REVIEW

File: tools/src/main/scala/scala/scalanative/build/Discover.scala [hunk 1]
```diff
@@ -150,7 +150,8 @@ object Discover {
 
     val command: Seq[String] = {
       if (Platform.isWindows) {
-        val arg = binPath.fold(binaryName)(p => s"$p:$binaryName")
+        val binName = s"$binaryName.exe"
```
Comment (ekrich) [inline tools/src/main/scala/scala/scalanative/build/Discover.scala:153]: `val binName = s"${binaryName}.exe"`

Comment (ekrich) [inline tools/src/main/scala/scala/scalanative/build/Discover.scala:153]:
```suggestion
        val binName = s"${binaryName}.exe"
```

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,5 @@
+# vim
+*.swp
```
Comment (ekrich) [inline .gitignore:2]: Do you mind moving this down to the bottom with the organized IDE/Editor ignores? The top part is pretty ugly.

**Review by ekrich — CHANGES_REQUESTED** [review]
Looks great, thanks for contributing. I guess I should have tested in other than `cmd` or Powershell 😄

File: .gitignore (hunk 2 above)
Comment (philwalk) [reply .gitignore]: done

**Comment by philwalk** [comment]
FYI, In order to fix problems with discovery of `clang++` in `cygwin`, I had to replace the installed symlink.

Originally it was a Windows `<JUNCTION>` file (not visible to java), and even after replacing that with a '<SYMLINK>', java was not finding it.   I experimented with `java.nio.file` symlink detection and de-referencing, but it turned out to be much simpler to just rename the symlink from `clang++` to `clang++.exe`.

**Review by WojciechMazur — APPROVED** [review]

**Comment by ekrich** [comment]
@philwalk This already is merged but if you want to do another PR and add a small paragraph under the Windows section for Cygwin and MSys with your workaround and any other install tips etc. for those environments that would be great. https://github.com/scala-native/scala-native/blob/master/docs/user/setup.rst

You can build the docs by running `make html` in the docs directory. You need Python 3 and pip to install sphinx, and recommonmark. We should have a section in the contributors section on what to run to get docs working but alas we don't.

**Comment by philwalk** [comment]
@ekrich - thanks, hopefully I can work on this again soon.

**Pull request merged**